File: app/src/main/java/com/paradoxo/amadeus/util/Preferencias.java
package com.paradoxo.amadeus.util;

import android.content.Context;
import android.content.SharedPreferences;

import static android.content.Context.MODE_PRIVATE;

public class Preferencias {
    private static final String PREFS_USU = "PrefsUsu";
    public static final String PREF_JA_FOI_ABERTO = "ja_foi_aberto";

    public static void confirmarAberturaApp(Context context) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(PREFS_USU, MODE_PRIVATE);
        SharedPreferences.Editor mEditor = sharedPreferences.edit();
        mEditor.putBoolean(PREF_JA_FOI_ABERTO,true);
        mEditor.apply();
    }

    public static boolean appJaFoiAberto(Context context) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(PREFS_USU, MODE_PRIVATE);
        return sharedPreferences.getBoolean(PREF_JA_FOI_ABERTO, false);
    }

    public static void setPrefBool(String nomeShared, boolean valor, Context context) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(PREFS_USU, MODE_PRIVATE);
        SharedPreferences.Editor mEditor = sharedPreferences.edit();
        mEditor.putBoolean(nomeShared, valor);
        mEditor.apply();
    }

    public static boolean getPrefBool(String nomePref, Context context, boolean valorPadrao) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(PREFS_USU, MODE_PRIVATE);
        return sharedPreferences.getBoolean(nomePref, valorPadrao);
    }

    public static void setPrefString(String conteudoShared, String nomeShared, Context context) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(PREFS_USU, MODE_PRIVATE);
        SharedPreferences.Editor mEditor = sharedPreferences.edit();
        mEditor.putString(nomeShared, conteudoShared);
        mEditor.apply();
    }

    public static String getPrefString(String nomeShared, Context context) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(PREFS_USU, MODE_PRIVATE);
        return sharedPreferences.getString(nomeShared, "");
    }

    public static Long getPrefLong(String nomeShared, Context context) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(PREFS_USU, MODE_PRIVATE);
        return sharedPreferences.getLong(nomeShared, 0);
    }

    public static void setPrefLong(String nomeShared, long valorShared, Context context) {
        SharedPreferences sharedPreferences = context.getSharedPreferences(PREFS_USU, MODE_PRIVATE);
        SharedPreferences.Editor mEditor = sharedPreferences.edit();
        mEditor.putLong(nomeShared, valorShared);
        mEditor.apply();
    }

}

File: app/src/main/java/com/paradoxo/amadeus/util/Preferencias.kt
package com.paradoxo.amadeus.util

import android.content.Context
import android.content.Context.MODE_PRIVATE

object Preferencias {
    private const val PREFS_USU = "PrefsUsu"
    const val PREF_JA_FOI_ABERTO = "ja_foi_aberto"

    fun confirmarAberturaApp(context: Context) {
        context.getSharedPreferences(PREFS_USU, MODE_PRIVATE).edit()
            .putBoolean(PREF_JA_FOI_ABERTO, true)
            .apply()
    }

    fun appJaFoiAberto(context: Context): Boolean =
        context.getSharedPreferences(PREFS_USU, MODE_PRIVATE)
            .getBoolean(PREF_JA_FOI_ABERTO, false)

    fun setPrefBool(nomeShared: String, valor: Boolean, context: Context) {
        context.getSharedPreferences(PREFS_USU, MODE_PRIVATE).edit()
            .putBoolean(nomeShared, valor)
            .apply()
    }

    fun getPrefBool(nomePref: String, context: Context, valorPadrao: Boolean): Boolean =
        context.getSharedPreferences(PREFS_USU, MODE_PRIVATE)
            .getBoolean(nomePref, valorPadrao)

    fun setPrefString(conteudoShared: String, nomeShared: String, context: Context) {
        context.getSharedPreferences(PREFS_USU, MODE_PRIVATE).edit()
            .putString(nomeShared, conteudoShared)
            .apply()
    }

    fun getPrefString(nomeShared: String, context: Context): String =
        context.getSharedPreferences(PREFS_USU, MODE_PRIVATE)
            .getString(nomeShared, "") ?: ""

    fun getPrefLong(nomeShared: String, context: Context): Long =
        context.getSharedPreferences(PREFS_USU, MODE_PRIVATE)
            .getLong(nomeShared, 0)

    fun setPrefLong(nomeShared: String, valorShared: Long, context: Context) {
        context.getSharedPreferences(PREFS_USU, MODE_PRIVATE).edit()
            .putLong(nomeShared, valorShared)
            .apply()
    }
}
